File: 1254-deepest-leaves-sum/deepest-leaves-sum.java
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     int val;
 *     TreeNode left;
 *     TreeNode right;
 *     TreeNode() {}
 *     TreeNode(int val) { this.val = val; }
 *     TreeNode(int val, TreeNode left, TreeNode right) {
 *         this.val = val;
 *         this.left = left;
 *         this.right = right;
 *     }
 * }
 */
class Solution {
    public int deepestLeavesSum(TreeNode root) {
            
            if(root==null){
                return 0;
            }
            List<TreeNode> queue=new ArrayList<>();
            queue.add(root);
            int sum=0;
            while(!queue.isEmpty()){
                int currentLevel= queue.size();
                sum=0;
                for(int i=0;i<currentLevel;i++){
                    TreeNode node=queue.removeFirst();
                    sum+=node.val;
                    if(node.left!=null){queue.add(node.left);}
                    if(node.right!=null){queue.add(node.right);}
                }
            }
            return sum;
        }
}

File: 1254-deepest-leaves-sum/deepest-leaves-sum.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right
from collections import deque
class Solution:
    def deepestLeavesSum(self, root: Optional[TreeNode]) -> int:
        if not root:
            return 0
        queue=deque([root])

        while queue:
            current_level=len(queue)
            current_level_sum=0
            for i in range(current_level):

                node=queue.popleft()
                current_level_sum+=node.val
                if node.left:
                    queue.append(node.left)
                if node.right:
                    queue.append(node.right)
        return current_level_sum

            

